src/components/views/rooms/RoomHeader.tsx

```diff
@@ -48,22 +51,287 @@ import { BetaPill } from "../beta/BetaCard";
+                    ev.preventDefault();
+                    startElementCall();
+                },
+                disabled: false,
+            };
+        } else {
+            return {
+                onClick: async (ev: ButtonEvent) => {
+                    ev.preventDefault();
+                    openMenu();
+                },
+                disabled: false,
+            };
+        }
+    }, [behavior, startLegacyCall, startElementCall, openMenu]);
+
+    const onJitsiClick = useCallback(async (ev: ButtonEvent) => {
+        ev.preventDefault();
+        closeMenu();
+        await startLegacyCall();
+    }, [closeMenu, startLegacyCall]);
+
+    const onElementClick = useCallback((ev: ButtonEvent) => {
+        ev.preventDefault();
+        closeMenu();
+        startElementCall();
+    }, [closeMenu, startElementCall]);
+
+    let menu: JSX.Element | null = null;
+    if (menuOpen) {
+        const buttonRect = buttonRef.current!.getBoundingClientRect();
+        menu = <IconizedContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu}>
+            <IconizedContextMenuOptionList>
+                <IconizedContextMenuOption label={_t("Video call (Jitsi)")} onClick={onJitsiClick} />
+                <IconizedContextMenuOption label={_t("Video call (Element Call)")} onClick={onElementClick} />
+            </IconizedContextMenuOptionList>
+        </IconizedContextMenu>;
+    }
+
+    return <>
+        <AccessibleTooltipButton
+            inputRef={buttonRef}
+            className="mx_RoomHeader_button mx_RoomHeader_videoCallButton"
+            onClick={onClick}
+            title={_t("Video call")}
+            tooltip={tooltip ?? _t("Video call")}
+            disabled={disabled || busy}
+        />
+        { menu }
+    </>;
+};
+
+interface CallButtonsProps {
+    room: Room;
+}
+
+// The header buttons for placing calls have become stupidly complex, so here
+// they are as a separate component
+const CallButtons: FC<CallButtonsProps> = ({ room }) => {
+    const [busy, setBusy] = useState(false);
+    const showButtons = useSettingValue<boolean>("showCallButtonsInComposer");
+    const groupCallsEnabled = useFeatureEnabled("feature_group_calls");
+    const videoRoomsEnabled = useFeatureEnabled("feature_video_rooms");
+    const isVideoRoom = useMemo(() => videoRoomsEnabled && calcIsVideoRoom(room), [videoRoomsEnabled, room]);
+    const useElementCallExclusively = useMemo(() => SdkConfig.get("element_call").use_exclusively, []);
+
+    const hasLegacyCall = useEventEmitterState(
+        LegacyCallHandler.instance,
+        LegacyCallHandlerEvent.CallsChanged,
+        useCallback(() => LegacyCallHandler.instance.getCallForRoom(room.roomId) !== null, [room]),
+    );
+
+    const widgets = useWidgets(room);
+    const hasJitsiWidget = useMemo(() => widgets.some(widget => WidgetType.JITSI.matches(widget.type)), [widgets]);
+
+    const hasElementCall = useCall(room.roomId) !== null;
```

Comment:
```suggestion
    const hasCall = useCall(room.roomId) !== null;
```

Since I think this also includes Jitsi video rooms